### DISCUSSION TITLE
Export docs to PDF

### DISCUSSION THREAD

**Discussion by iozcelik** [withastro/starlight#964] [Feature Requests]
### What version of `starlight` are you using?

0.10.2

### What is your idea?

After writing documentation, sometimes we need to export it as pdf. So, an export future will be good.

### Why is this feature necessary?

It will be use for offline documentation.

### Do you have examples of this feature in other projects?

-DocFx
-[docusaurus-prince-pdf](https://github.com/signcl/docusaurus-prince-pdf)

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by delucis** [comment]
This sounds really cool! I'm not sure whether this is something we'll be able to support as a core part of Starlight, but could be a good opportunity for a [community tool](https://starlight.astro.build/showcase/#community-plugins)? (Similar to the Docusaurus example you shared which is not part of the main Docusaurus project.)

**Comment by codingluke** [comment]
Hey there. I just wanted to point to https://github.com/jean-humann/docs-to-pdf which is actually a generic solution to paginate through a docs page, following a "next" link. It is a hard fork from mr-pdf. Here an example on [how to integrate](https://github.com/codingluke/bbzbl-modul-319/blob/main/build-pdf.mjs) in a [ci pipeline](https://github.com/codingluke/bbzbl-modul-319/blob/main/.github/workflows/deploy.yml#L30-L31), and the [result](https://codingluke.github.io/bbzbl-modul-319/assets/pdf/bbzbl-modul-319.pdf). There is still a very important feature missing to [generate an outline](https://github.com/jean-humann/docs-to-pdf/issues/131). 

Also the vitepress guys seem also to do great things in this direction. https://github.com/condorheroblog/html-export-pdf-cli/

Would be great to see synergy instead of parallel development 🚀

**Comment by adam4nj** [comment]
Have you tried using [md-to-pdf.fly.dev](https://md-to-pdf.fly.dev/)

**Reply by abdulhakim-alshanqiti** [reply]
it uses pandoc so shouldn't we use pandoc insted?

**Comment by ailequal** [comment]
I ended up using [simonhaenisch/md-to-pdf](https://github.com/simonhaenisch/md-to-pdf) with this command `md-to-pdf --basedir "$BASEDIR" ./file.md` where `BASEDIR` is the root path of your starlight project. In this way any image that has been used will be included in the final PDF.

You can also add pagination and a title to the document and other things, check the project documentation.

In my case, I have written a bash script that automates some steps and I do quite like this solution in the end, even if not "native".

**Reply by abdulhakim-alshanqiti** [reply]
for me i decided to go the pwa route insted of turning the website into a pdf

**Comment by Linkerin** [comment]
Hello there. I recently stumbled upon the same need and built a tool for generating PDF from Starlight docs. It's not polished yet, nevertheless, it works as expected. Check it out: [starlight-to-pdf](https://github.com/Linkerin/starlight-to-pdf).

**Reply by abdulhakim-alshanqiti** [reply]
Thank you so much!!!

**Reply by delucis** [reply]
Awesome stuff! 🌟

**Reply by Syntarex** [reply]
Love you for that! Thanks

**Comment by HiDeoo** [comment]
Considering many suggestions have been listed to handle this use case, either with CLI tool or [Starlight plugins](https://github.com/Linkerin/starlight-to-pdf), it doesn't make a lot of sense to add this feature to Starlight core. Considering this, I'm going to close this issue for now.

**Comment by david-manm** [comment]
Disclosure: I build manuscrio.com It’s commercial, with free licences for open-source projects.

Manuscrio focuses on publication-quality manuals:
 
  - Exports PDF directly from dist/, locally or in CI—no running site required.
  - Produces numbered toc with page numbers and PDF bookmarks.
  - Handles wide tables, repeated headers, and long code without overflowing the page.

  It also supports other documentation frameworks.

  Starlight example and CI workflow: https://github.com/manuscrio/starlight-pdf-export
